domain/src/Commands/DomainCommands.php

```diff
@@ -0,0 +1,1288 @@
+      }
+      else {
+        $this->logger()->info(dt('!domain is already enabled.',
+          ['!domain' => $domain->getHostname()]));
+      }
+      if ($domain->status()) {
+        return 'enabled';
+      }
+      else {
+        return 'disabled';
+      }
+    }
+    return 'unknown domain';
+  }
+
+  /**
+   * Changes a domain label.
+   *
+   * @param $domain_id
+   *   The numeric id or hostname of the domain to relabel.
+   * @param $name
+   *   The name to use for the domain.
+   * @usage drush domain-name example.com Foo
+   * @usage drush domain-name 1 Foo
+   *
+   * @command domain:name
+   * @aliases domain-name
+   *
+   * @throws \Drupal\domain\Commands\DomainCommandException
+   */
+  public function renameDomain($domain_id, $name) {
+    // Resolve the domain.
+    if ($domain = $this->getDomainFromArgument($domain_id)) {
+      $domain->saveProperty('name', $name);
+      return $domain->name();
+    }
+    return 'unknown domain';
+  }
+
+  /**
+   * Changes a domain name machine ID.
+   *
+   * This is disabled because we need to modify all content in a 'reassign
+   * domains' type way.
+   *
+   * @param $domain_id
+   *   The numeric id or hostname of the domain to rename.
+   * @param $machine_name
+   *   The machine-readable name to use for the domain.
+   * @usage drush domain-machine-name example.com foo
+   * @usage drush domain-machine-name 1 foo
+   *
+   * @command domain:machine-name
+   * @aliases domain-machine-name
+   *
+   * @throws \Drupal\domain\Commands\DomainCommandException
+   */
+//  public function machineName($domain_id, $machine_name) {
+//    $domain_storage = $this->getStorage();
+//    $machine_name = $domain_storage->createMachineName($machine_name);
+//    // Resolve the domain.
+//    if ($domain = $this->getDomainFromArgument($domain_id)) {
+//      $results = $domain_storage->loadByProperties(['machine_name' => $machine_name]);
+//      foreach ($results as $result) {
+//        if ($result->id() == $machine_name) {
+//          $this->logger()->warning(dt('The machine_name !machine_name is being used by domain !hostname.',
+//            ['!machine_name' => $machine_name, '!hostname' => $result->getHostname()]));
+//          return;
+//        }
+//      }
+//      $domain->saveProperty('id', $machine_name);
+//    }
+//  }
+
+  /**
+   * Changes a domain scheme.
+   *
+   * @param $domain_id
+   *   The numeric id or hostname of the domain to change.
+   * @param $scheme (deprecated)
+   *   The URL schema (http or https) to use for the domain.
+   * @param array $options
+   *    An associative array of options whose values come from cli, aliases,
+   *    config, etc.
+   *
+   * @usage drush domain-scheme example.com --https
+   * @usage drush domain-scheme 681 https
+   *
+   * @option http
+   *   Set the domain access scheme to http (no SSL).
+   * @option https
+   *   Set the domain access scheme to https (with SSL).
+   *
```

Comment:
There is a third option now: `variable`, which `Sets the domain's URL scheme to inherit from the request.`

Reply:
I have added the third option and tidied up the code somewhat as well; the old "drush scheme http" is now "drush domain:scheme --set=http" as I feel this is a lot clearer what is happening. I have also modified and updated the code relating to scheme in the domain:add command.

Reply:
A thought: should there be a standard function somewhere that answers the question "Is this string a valid scheme name"?

Reply:
I don't think so. We only support `http` and `https` so those are hardcoded in most places. Even here, the input is numeric and we convert to a fixed string.